### ISSUE TITLE
Unable to send JSON list using @json annotation

### ISSUE THREAD

**Issue by kyrrigle** [prkumar/uplink#103]
I needed to post a JSON list and ran into difficulty using @json and Body.  The code seems to assume that a json body will always be a dictionary and tries to loop over the items.  Specifically, in decorators.py the `old_body` will be a list instead of the expected dictionary.

```
    @classmethod
    def set_json_body(cls, request_builder):
        body = request_builder.info.setdefault("json", {})
        old_body = request_builder.info.pop("data", {})
        for path in old_body:
            if isinstance(path, tuple):
                cls._sequence_path_resolver(path, old_body[path], body)
            else:
                body[path] = old_body[path]
```

As a workaround I found that I could just pass `json.dumps` as a converter-

```
    @headers({"Content-Type": "application/json"})
    @post("/collections/{name}/documents", args={"name": Path, "documents": Body(json.dumps)})
    def send_documents(self, name, documents):
        '''Send some documents'''
```

and then things will work

```
data = ["one", "two", "three"]
service.send_documents("testing", data)
```

**Comment by prkumar** [comment]
@kyrrigle, Sorry about that! Fortunately, this should be a quick fix, which I'm tracking with #104.

Here's a little background, in case you're interested: the loop in `set_json_body` is mainly responsible for resolving any nested JSON field references. Namely, such a reference is possible with the `Field` annotation. For example, consider a Update User API endpoint that expects a JSON body in the following format:

```
{
    "user": {
        "name": "username"
    }
}
```

We could send a username update request using the following method (note the argument of `Field` is a tuple):

```
@json
@post("/user", args={"name": Field(("user", "name")})
def update_username(self, name):
    """Update the current user's username"""
```

Finally, when processing the request body, the `set_json_body` method loops over the body to resolve these JSON field references. 

Anyways, the fix I'm working on would skip the loop if the body is not an instance of `collections.Mapping`. This change should be live in a couple of hours, barring any complications. If you want to try it out beforehand, you can install the branch directly using `pip`:

```
$ pip install https://github.com/prkumar/uplink/archive/hotfix/v0.5.5/json-list-as-body.zip
```

Thanks again!

**Comment by prkumar** [comment]
@kyrrigle, the fix is included in v0.5.5 release. When you get the chance, please upgrade. Thanks for filing a ticket.
